File: lib/external_api.rb
require 'json'

class ExternalApi
  PAGE_SIZE = 10
  def initialize(db_file)
    @db = JSON.parse(File.read(db_file)).group_by{|record| record["evaluado_id"].to_i}
  end

  def fetch_results(employee_id, page = 1)
    # Return the next page of results.  If page == 0, starts from the
    # beginning.  Otherwise, fetches the next 10 records after the last page.
    # returns:
    #     {
    #         "results": [{evaluado: text, id: integer, result: integer}, {evaluado: text, id: integer, result: integer}, ...],
    #         "page": 1
    #     }
    return {'page': page, 'results': nil}.to_json if page.negative? || page.zero? || !db.key?(employee_id)
    return {
      'page': page,
      'results': db.fetch(employee_id, nil)&.slice(PAGE_SIZE * (page - 1), PAGE_SIZE),
    }.to_json
  end

  private
    attr_reader :db
end

File: ts/src/external_api.ts
import data from '../data/example.json';

export type Result = {
  id: number,
  evaluado_id: number,
  result: number,
  kind: string
}

export class ExternalApi {
  db: Object;
  pageSize: number;

  constructor() {
    this.pageSize = 10;
    let tmp = (data as Result[]);
    this.db = {};
    tmp.forEach((item) => {
      if(!(item.evaluado_id in this.db)) {
        this.db[item.evaluado_id] = new Array()
      }
      this.db[item.evaluado_id].push(item)
    });
  }

  fetch_participants() {
    return Object.keys(this.db);
  }

  fetch_results(employee_id: string, page:number= 1) {
    /*
     * returns {"page": number, "results": [{"id": number, "evaluado_id": number, "result": number, "kind": string}]}
     */
    if(page <= 0 || !(employee_id in this.db)) {
      return JSON.stringify({page: page, results: null});
    }
    let start: number = this.pageSize * (page - 1);
    let end: number = start + this.pageSize;
    return JSON.stringify(
      {
        page: page,
        results: this.db[employee_id]?.slice(start, end)
      }
    );
  }
}
